mobile/lib/services/search_service.dart

```diff
@@ -1413,10 +1414,30 @@ class SearchService {
     int? limit,
   ) async {
     try {
+      final int ownerID = Configuration.instance.getUserID()!;
       final searchResults = <GenericSearchResult>[];
       final allFiles = await getAllFilesForSearch();
       final peopleToSharedFiles = <User, List<EnteFile>>{};
+      final peopleToSharedAlbums = <String, List<Collection>>{};
       final existingEmails = <String>{};
+      final familyEmails = UserService.instance.getEmailIDsOfFamilyMember();
+      final List<Collection> collections = _collectionService
+          .getCollectionsForUI(includedShared: true, includeCollab: true);
+
+      for (Collection collection in collections) {
+        if (collection.isHidden() ||
+            collection.isArchived() ||
+            collection.isOwner(ownerID)) {
+          continue;
+        }
+        if (collection.owner.id != ownerID) {
```

Comment:
Isn't this check unnecessary since `collection.isOwner(ownerID)` is already checked in the `if` condition above?